src/manual/peer2peer/js/main.js

```diff
@@ -249,13 +247,13 @@ function updateGetUserMediaConstraints() {
 
     if ($('audio').checked) {
       if (devices.audioId !== null) {
-        global.constraints.audio = {optional: [{deviceId: devices.audioId}]};
+        global.constraints.audio = {deviceId: devices.audioId};
       }
     }
 
     if ($('video').checked) {
       if (devices.videoId !== null) {
-        global.constraints.video.optional.push({deviceId: devices.videoId});
+        global.constraints.video.deviceId = devices.videoId;
```

Comment:
I did not consider the code around it, I've now made the video constraints format the same as audio.